src/composites/TreeDisplay/nodeDescriptionPanel/NodeDescriptionPanel.tsx

```diff
@@ -2,34 +2,75 @@ import "./NodeDescriptionPanel.css"
 import {useEffect, useState} from "react";
 import {determineNodeInfo} from "./NodeDescriptionPanelHelpers";
 
-export default function NodeDescriptionPanel(props) {
+export default function NodeDescriptionPanel(props: { nodes: any[]; impacts: any }) {
+  const [nodes, setNodes] = useState<any[]>([]);
+  const [orderBy, setOrderBy] = useState<string>("default");
+  const [orderDirection, setOrderDirection] = useState<string>("asc");
 
-    const [nodes,setNodes] = useState([]);
+  // Rerender this component whenever the nodes prop from TreeDisplay changes.
+  useEffect(() => {
+    setNodes(props.nodes);
+  }, [props.nodes]);
 
-    // Rerender this component whenever the nodes prop from TreeDisplay changes.
-    useEffect(() => {
-        setNodes(props.nodes)
-    },[props.nodes])
+  function makeNodePanelRectangles() {
```

Comment:
I would extract this outside the component. Defining this here will cause this function to be redefined on re-render, and this seems like a reusable piece of code we can potentially use in other components.

Reply:
In addition, I would wrap this call in a useMemo(), so that we don't re-sort on re-render.

Reply:
wrapped in useMemo, function is still in component for now.